docs/inference.md

```diff
@@ -0,0 +1,31 @@
+---
+title: 🤗 Hub Inference API docs
+---
+
+<h1>🤗 Hub Inference API</h1>
+
+For detailed usage documentation, please refer to [Accelerated Inference API Documentation](https://api-inference.huggingface.co/docs/python/html/index.html).
+
+
+## What technology do you use to power the inference API?
+
+For 🤗 Transformers models, the API is built on top of our [Pipelines](https://huggingface.co/transformers/main_classes/pipelines.html) feature.
+
+On top of `Pipelines` and depending on the model type, we build a number of production optimizations like:
+- compiling models to optimized intermediary representations (e.g. [ONNX](https://medium.com/microsoftazure/accelerate-your-nlp-pipelines-using-hugging-face-transformers-and-onnx-runtime-2443578f4333)),
+- maintaining a Least Recently Used cache ensuring that the most popular models are always loaded,
+- scaling the underlying compute infrastructure on the fly depending on the load constraints.
```

Comment:
This explains for 🤗 Transformers, but not for others - I think it would help to explain a bit how it's currently done for other libraries, for example by linking to library-specific code samples: https://github.com/huggingface/huggingface_hub/blob/main/api-inference-community/docker_images/sentence_transformers/app/pipelines/sentence_similarity.py

Explaining that it's all dockerized and uses `starlette` under the hood would also help clarify things.

Reply:
Done. Added below